lib/utils/useWeb3Modal/index.tsx

```diff
@@ -29,52 +28,112 @@ export const useWeb3Modal = (strings: Web3ModalStrings): Web3ModalState => {
+        const address = await signer.getAddress();
+        const network = await provider.getNetwork();
+        const newState: ConnectedWeb3State = {
+          provider,
+          signer,
+          address,
+          network,
+          isConnected: true,
+        };
+        setWeb3State(newState);
+      } else if (wallet === "walletConnect") {
+        const walletConnectProvider = new WalletConnectProvider({
+          rpc: { 137: urls.polygonMainnetRpc },
+        });
+        await walletConnectProvider.enable();
+        const provider = new ethers.providers.Web3Provider(
+          walletConnectProvider
+        ) as unknown as TypedProvider;
+        const signer = await provider.getSigner();
+        const address = await signer.getAddress();
+        const network = await provider.getNetwork();
+        const newState: ConnectedWeb3State = {
+          provider,
+          signer,
+          address,
+          network,
+          isConnected: true,
+        };
+        setWeb3State(newState);
+      } else if (wallet === "torus") {
+        const torus = new Torus();
+        await torus.init({
+          network: {
+            host: "matic",
+            chainId: 137,
+            networkName: "Polygon",
+          },
+        });
+        await torus.login();
+        const provider = new ethers.providers.Web3Provider(
+          torus.provider
+        ) as unknown as TypedProvider;
+        const signer = await provider.getSigner();
+        const address = await signer.getAddress();
+        const network = await provider.getNetwork();
+        const newState: ConnectedWeb3State = {
+          provider,
+          signer,
+          address,
+          network,
+          isConnected: true,
+        };
+        setWeb3State(newState);
+      } else {
+        throw new Error("error connecting");
+      }
+    } catch (e: any) {
+      console.log("error connecting:", e);
+      return e;
     }
-    const wrappedProvider = await modalRef.current.connect();
-    const provider = new providers.Web3Provider(
-      wrappedProvider
-    ) as unknown as TypedProvider; // assert for better typings, see event handlers below
-    const signer = provider.getSigner();
-    const address = await signer.getAddress();
-    const network = await provider.getNetwork();
-    const newState: ConnectedWeb3State = {
-      provider,
-      signer,
-      address,
-      network,
-      isConnected: true,
-    };
-    setWeb3State(newState);
   };
 
-  // Auto download dependencies and connect to the cached provider
+  // Auto connect to the cached provider if its metamask
+  // brave injects window.ethereum as well
   useEffect(() => {
-    const init = async () => {
-      if (!modalRef.current) {
-        modalRef.current = await downloadModal();
-      }
-      if (modalRef.current.cachedProvider) {
-        connect();
-      }
-    };
-    init();
+    if (window.ethereum) {
+      connect();
```

Comment:
We need to maintain the behavior that if a user connects and refreshes the page, it should auto reconnect to the same wallet.

If they disconnect, and refresh, it should not try to connect again.

So we can't rely on `window.ethereum` as 1. that only works for `metamask`/`brave`, and 2. it's always there even after they disconnect.
We probably need to maintain provider info in `LocalStorage`. I wonder if we can just write the entire ethers `provider` object to localstorage... might be worth looking how web3modal does this under the hood?

Reply:
Fixed. We are now storing a web3-wallet value in localStorage that is used to reconnect a user on refresh. web3 modal also does this.